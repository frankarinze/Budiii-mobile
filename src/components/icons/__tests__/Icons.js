import React from 'react';
import renderer from 'react-test-renderer';

import SvgCircleLeft from '../Svg.CircleLeft';
import SvgCog from '../Svg.Cog';
import SvgDiscover from '../SvgDiscover';
import SvgMessages from '../SvgMessages';
import SvgStats from '../Svg.Stats';

describe('SVG Icons', () => {
  it('<SvgCircleLeft /> renders correctly', () => {
    const tree = renderer.create(<SvgCircleLeft />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('<SvgCog /> renders correctly', () => {
    const tree = renderer.create(<SvgCog />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('<SvgDiscover /> renders correctly', () => {
    const tree = renderer.create(<SvgDiscover />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('<SvgMessages /> renders correctly', () => {
    const tree = renderer.create(<SvgMessages />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('<SvgStats /> renders correctly', () => {
    const tree = renderer.create(<SvgStats />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
