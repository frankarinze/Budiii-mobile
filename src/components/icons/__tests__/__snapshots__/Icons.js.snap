// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SVG Icons <SvgCircleLeft /> renders correctly 1`] = `
<RNSVGSvgView
  align="xMidYMid"
  bbHeight="24"
  bbWidth="24"
  height={24}
  meetOrSlice={0}
  minX={0}
  minY={0}
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
        "borderWidth": 0,
      },
      undefined,
      false,
      Object {
        "flex": 0,
        "height": 24,
        "width": 24,
      },
    ]
  }
  vbHeight={32}
  vbWidth={32}
  width={24}
>
  <RNSVGGroup
    fill={
      Array [
        0,
        0,
        0,
        0,
        1,
      ]
    }
    fillOpacity={1}
    fillRule={1}
    font={Object {}}
    matrix={
      Array [
        1,
        0,
        0,
        1,
        0,
        0,
      ]
    }
    opacity={1}
    originX={0}
    originY={0}
    propList={Array []}
    rotation={0}
    scaleX={1}
    scaleY={1}
    skewX={0}
    skewY={0}
    stroke={null}
    strokeDasharray={null}
    strokeDashoffset={null}
    strokeLinecap={0}
    strokeLinejoin={0}
    strokeMiterlimit={4}
    strokeOpacity={1}
    strokeWidth="1"
    x={0}
    y={0}
  >
    <RNSVGPath
      d="M16 32c8.837 0 16-7.163 16-16S24.837 0 16 0 0 7.163 0 16s7.163 16 16 16zm0-29c7.18 0 13 5.82 13 13s-5.82 13-13 13S3 23.18 3 16 8.82 3 16 3z"
      fill={
        Array [
          0,
          0.8156862745098039,
          0.8,
          0.8156862745098039,
          1,
        ]
      }
      fillOpacity={1}
      fillRule={1}
      matrix={
        Array [
          1,
          0,
          0,
          1,
          0,
          0,
        ]
      }
      opacity={1}
      originX={0}
      originY={0}
      propList={
        Array [
          "fill",
        ]
      }
      rotation={0}
      scaleX={1}
      scaleY={1}
      skewX={0}
      skewY={0}
      stroke={null}
      strokeDasharray={null}
      strokeDashoffset={null}
      strokeLinecap={0}
      strokeLinejoin={0}
      strokeMiterlimit={4}
      strokeOpacity={1}
      strokeWidth="1"
      x={0}
      y={0}
    />
    <RNSVGPath
      d="M20.914 9.914l-2.829-2.829-8.914 8.914 8.914 8.914 2.828-2.828-6.086-6.086z"
      fill={
        Array [
          0,
          0.8156862745098039,
          0.8,
          0.8156862745098039,
          1,
        ]
      }
      fillOpacity={1}
      fillRule={1}
      matrix={
        Array [
          1,
          0,
          0,
          1,
          0,
          0,
        ]
      }
      opacity={1}
      originX={0}
      originY={0}
      propList={
        Array [
          "fill",
        ]
      }
      rotation={0}
      scaleX={1}
      scaleY={1}
      skewX={0}
      skewY={0}
      stroke={null}
      strokeDasharray={null}
      strokeDashoffset={null}
      strokeLinecap={0}
      strokeLinejoin={0}
      strokeMiterlimit={4}
      strokeOpacity={1}
      strokeWidth="1"
      x={0}
      y={0}
    />
  </RNSVGGroup>
</RNSVGSvgView>
`;

exports[`SVG Icons <SvgCog /> renders correctly 1`] = `
<RNSVGSvgView
  align="xMidYMid"
  bbHeight="20"
  bbWidth="20"
  height={20}
  meetOrSlice={0}
  minX={0}
  minY={0}
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
        "borderWidth": 0,
      },
      undefined,
      false,
      Object {
        "flex": 0,
        "height": 20,
        "width": 20,
      },
    ]
  }
  vbHeight={32}
  vbWidth={32}
  width={20}
>
  <RNSVGGroup
    fill={
      Array [
        0,
        0,
        0,
        0,
        1,
      ]
    }
    fillOpacity={1}
    fillRule={1}
    font={Object {}}
    matrix={
      Array [
        1,
        0,
        0,
        1,
        0,
        0,
      ]
    }
    opacity={1}
    originX={0}
    originY={0}
    propList={Array []}
    rotation={0}
    scaleX={1}
    scaleY={1}
    skewX={0}
    skewY={0}
    stroke={null}
    strokeDasharray={null}
    strokeDashoffset={null}
    strokeLinecap={0}
    strokeLinejoin={0}
    strokeMiterlimit={4}
    strokeOpacity={1}
    strokeWidth="1"
    x={0}
    y={0}
  >
    <RNSVGPath
      d="M29.181 19.07c-1.679-2.908-.669-6.634 2.255-8.328l-3.145-5.447a6.022 6.022 0 0 1-3.058.829c-3.361 0-6.085-2.742-6.085-6.125h-6.289a6.023 6.023 0 0 1-.811 3.07C10.369 5.977 6.637 6.966 3.709 5.28L.565 10.727a6.023 6.023 0 0 1 2.246 2.234c1.676 2.903.672 6.623-2.241 8.319l3.145 5.447a6.022 6.022 0 0 1 3.044-.82c3.35 0 6.067 2.725 6.084 6.092h6.289a6.032 6.032 0 0 1 .811-3.038c1.676-2.903 5.399-3.894 8.325-2.219l3.145-5.447a6.032 6.032 0 0 1-2.232-2.226zM16 22.479A6.48 6.48 0 1 1 16 9.52a6.48 6.48 0 0 1 0 12.959z"
      fill={
        Array [
          0,
          0.8156862745098039,
          0.8,
          0.8156862745098039,
          1,
        ]
      }
      fillOpacity={1}
      fillRule={1}
      matrix={
        Array [
          1,
          0,
          0,
          1,
          0,
          0,
        ]
      }
      opacity={1}
      originX={0}
      originY={0}
      propList={
        Array [
          "fill",
        ]
      }
      rotation={0}
      scaleX={1}
      scaleY={1}
      skewX={0}
      skewY={0}
      stroke={null}
      strokeDasharray={null}
      strokeDashoffset={null}
      strokeLinecap={0}
      strokeLinejoin={0}
      strokeMiterlimit={4}
      strokeOpacity={1}
      strokeWidth="1"
      x={0}
      y={0}
    />
  </RNSVGGroup>
</RNSVGSvgView>
`;

exports[`SVG Icons <SvgDiscover /> renders correctly 1`] = `
<RNSVGSvgView
  align="xMidYMid"
  bbHeight="20"
  bbWidth="20"
  height={20}
  meetOrSlice={0}
  minX={0}
  minY={0}
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
        "borderWidth": 0,
      },
      undefined,
      false,
      Object {
        "flex": 0,
        "height": 20,
        "width": 20,
      },
    ]
  }
  vbHeight={32}
  vbWidth={32}
  width={20}
>
  <RNSVGGroup
    fill={
      Array [
        0,
        0,
        0,
        0,
        1,
      ]
    }
    fillOpacity={1}
    fillRule={1}
    font={Object {}}
    matrix={
      Array [
        1,
        0,
        0,
        1,
        0,
        0,
      ]
    }
    opacity={1}
    originX={0}
    originY={0}
    propList={Array []}
    rotation={0}
    scaleX={1}
    scaleY={1}
    skewX={0}
    skewY={0}
    stroke={null}
    strokeDasharray={null}
    strokeDashoffset={null}
    strokeLinecap={0}
    strokeLinejoin={0}
    strokeMiterlimit={4}
    strokeOpacity={1}
    strokeWidth="1"
    x={0}
    y={0}
  >
    <RNSVGPath
      d="M32 18.451L16 6.031 0 18.451v-5.064L16 .967l16 12.42zM28 18v12h-8v-8h-8v8H4V18l12-9z"
      fill={
        Array [
          0,
          0.8156862745098039,
          0.8,
          0.8156862745098039,
          1,
        ]
      }
      fillOpacity={1}
      fillRule={1}
      matrix={
        Array [
          1,
          0,
          0,
          1,
          0,
          0,
        ]
      }
      opacity={1}
      originX={0}
      originY={0}
      propList={
        Array [
          "fill",
        ]
      }
      rotation={0}
      scaleX={1}
      scaleY={1}
      skewX={0}
      skewY={0}
      stroke={null}
      strokeDasharray={null}
      strokeDashoffset={null}
      strokeLinecap={0}
      strokeLinejoin={0}
      strokeMiterlimit={4}
      strokeOpacity={1}
      strokeWidth="1"
      x={0}
      y={0}
    />
  </RNSVGGroup>
</RNSVGSvgView>
`;

exports[`SVG Icons <SvgMessages /> renders correctly 1`] = `
<RNSVGSvgView
  align="xMidYMid"
  bbHeight="20"
  bbWidth="20"
  height={20}
  meetOrSlice={0}
  minX={0}
  minY={0}
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
        "borderWidth": 0,
      },
      undefined,
      false,
      Object {
        "flex": 0,
        "height": 20,
        "width": 20,
      },
    ]
  }
  vbHeight={32}
  vbWidth={32}
  width={20}
>
  <RNSVGGroup
    fill={
      Array [
        0,
        0,
        0,
        0,
        1,
      ]
    }
    fillOpacity={1}
    fillRule={1}
    font={Object {}}
    matrix={
      Array [
        1,
        0,
        0,
        1,
        0,
        0,
      ]
    }
    opacity={1}
    originX={0}
    originY={0}
    propList={Array []}
    rotation={0}
    scaleX={1}
    scaleY={1}
    skewX={0}
    skewY={0}
    stroke={null}
    strokeDasharray={null}
    strokeDashoffset={null}
    strokeLinecap={0}
    strokeLinejoin={0}
    strokeMiterlimit={4}
    strokeOpacity={1}
    strokeWidth="1"
    x={0}
    y={0}
  >
    <RNSVGPath
      d="M20 8V0H6L0 6v18h12v8h20V8H20zM6 2.828V6H2.828L6 2.828zM2 22V8h6V2h10v6l-6 6v8H2zm16-11.172V14h-3.172L18 10.828zM30 30H14V16h6v-6h10v20z"
      fill={
        Array [
          0,
          0.8156862745098039,
          0.8,
          0.8156862745098039,
          1,
        ]
      }
      fillOpacity={1}
      fillRule={1}
      matrix={
        Array [
          1,
          0,
          0,
          1,
          0,
          0,
        ]
      }
      opacity={1}
      originX={0}
      originY={0}
      propList={
        Array [
          "fill",
        ]
      }
      rotation={0}
      scaleX={1}
      scaleY={1}
      skewX={0}
      skewY={0}
      stroke={null}
      strokeDasharray={null}
      strokeDashoffset={null}
      strokeLinecap={0}
      strokeLinejoin={0}
      strokeMiterlimit={4}
      strokeOpacity={1}
      strokeWidth="1"
      x={0}
      y={0}
    />
  </RNSVGGroup>
</RNSVGSvgView>
`;

exports[`SVG Icons <SvgStats /> renders correctly 1`] = `
<RNSVGSvgView
  align="xMidYMid"
  bbHeight="20"
  bbWidth="20"
  height={20}
  meetOrSlice={0}
  minX={0}
  minY={0}
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
        "borderWidth": 0,
      },
      undefined,
      false,
      Object {
        "flex": 0,
        "height": 20,
        "width": 20,
      },
    ]
  }
  vbHeight={32}
  vbWidth={32}
  width={20}
>
  <RNSVGGroup
    fill={
      Array [
        0,
        0,
        0,
        0,
        1,
      ]
    }
    fillOpacity={1}
    fillRule={1}
    font={Object {}}
    matrix={
      Array [
        1,
        0,
        0,
        1,
        0,
        0,
      ]
    }
    opacity={1}
    originX={0}
    originY={0}
    propList={Array []}
    rotation={0}
    scaleX={1}
    scaleY={1}
    skewX={0}
    skewY={0}
    stroke={null}
    strokeDasharray={null}
    strokeDashoffset={null}
    strokeLinecap={0}
    strokeLinejoin={0}
    strokeMiterlimit={4}
    strokeOpacity={1}
    strokeWidth="1"
    x={0}
    y={0}
  >
    <RNSVGPath
      d="M0 26h32v4H0zm4-8h4v6H4zm6-8h4v14h-4zm6 6h4v8h-4zm6-12h4v20h-4z"
      fill={
        Array [
          0,
          0.8156862745098039,
          0.8,
          0.8156862745098039,
          1,
        ]
      }
      fillOpacity={1}
      fillRule={1}
      matrix={
        Array [
          1,
          0,
          0,
          1,
          0,
          0,
        ]
      }
      opacity={1}
      originX={0}
      originY={0}
      propList={
        Array [
          "fill",
        ]
      }
      rotation={0}
      scaleX={1}
      scaleY={1}
      skewX={0}
      skewY={0}
      stroke={null}
      strokeDasharray={null}
      strokeDashoffset={null}
      strokeLinecap={0}
      strokeLinejoin={0}
      strokeMiterlimit={4}
      strokeOpacity={1}
      strokeWidth="1"
      x={0}
      y={0}
    />
  </RNSVGGroup>
</RNSVGSvgView>
`;
